examples/atomiqueue_example.rs

```diff
@@ -64,22 +72,20 @@ fn main() {
 
     set_mtimecmp(&clint.mtime, &mut clint.mtimecmp);
     clint.mtimer.enable();
-    writeln!(Stdout(&mut tx), "PRE").unwrap();
-    unsafe {
-        interrupt::enable();
-    }
-    writeln!(Stdout(&mut tx), "POST").unwrap();
+
+    writeln!(stdout, "PRE").unwrap();
+    unsafe { interrupt::enable(); }
+    writeln!(stdout, "POST").unwrap();
 
     loop {
         if button_pin.is_low() {
-            if let Ok(Some(timestamp)) = queue.pop() {
-                writeln!(Stdout(&mut tx), "Time: {:?} has been popped!", timestamp).unwrap();
+            if let Ok(Some(timestamp)) = QUEUE.pop() {
+                writeln!(stdout, "Time: {:?} has been popped!", timestamp).unwrap();
             };
         }
 
-        if PUSHED.load(Ordering::Relaxed) {
-            PUSHED.store(false, Ordering::Relaxed);
-            writeln!(Stdout(&mut tx), "Timestamp has been pushed!").unwrap();
+        if PUSHED.compare_and_swap(true, false, Ordering::Relaxed) == false {
```

Comment:
This change causes the example to break:

![image](https://user-images.githubusercontent.com/9815664/52521882-fe6c2d00-2c32-11e9-9a01-16d4deaadc65.png)

The upper part of the console output here is what the program was producing as-is; I reverted these two line changes back to an explicit load and store instead of compare and swap, and the usual behavior can be seen at the bottom of the console output.